mountpoint-s3-crt/src/lib.rs

```diff
@@ -73,24 +74,37 @@ pub(crate) unsafe fn aws_byte_cursor_as_slice<'a>(cursor: &aws_byte_cursor) -> &
 pub(crate) trait CrtError: Sized {
     type Return;
 
-    /// Safety: This must only be used immediately on a pointer returned from the CRT, with no other
+    /// # Safety
+    /// This must only be used immediately on a pointer returned from the CRT, with no other
```

Comment:
Should this also mention that there shouldn't be any `await`s or similar points where a different task could be scheduled on the same thread/the current task gets rescheduled?